src/vos/vos_aggregate.c

```diff
@@ -1902,10 +1937,25 @@ vos_aggregate_post_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 		 * -DER_TX_BUSY error indicates current ilog aggregation
 		 * aborted on hitting uncommitted entry, this should be a very
 		 * rare case, we'd suppress the error here to keep aggregation
-		 * moving forward.
+		 * moving forward.   We do, however, need to ensure we do not
+		 * aggregate anything in the parent path.  Otherwise, we could
+		 * orphan the current entry due to incarnation log semantics.
```

Comment:
I don't quite see the "incarnation log semantics" here, could you explain a bit?

I don't see how this "ap_skip_obj/dkey/akey" works as well, the intention is to set the skip flag for current epoch range aggregation? Or for all epoch ranges aggregation?

Reply:
I discussed this with you offline but for everyone's benefit, the issue is that the incarnation log doesn't contain enough information to know at the key level that there are in-progress updates that we may orphan by aggregating the ilog.  The example I used is this one

1. committed update at 1
2. in progress update at 2 - this will not generate an incarnation log entry
3. punch at 3

From the ilog perspective, we can delete everything but the subtree will not be empty due to the inprogress update.